hazelcast-client-new/src/test/java/com/hazelcast/client/ClientConnectionTest.java

```diff
@@ -42,36 +41,37 @@
 
 import static org.junit.Assert.assertEquals;
 
-@RunWith(HazelcastSerialClassRunner.class)
+@RunWith(HazelcastParallelClassRunner.class)
 @Category(QuickTest.class)
 public class ClientConnectionTest extends HazelcastTestSupport {
 
+    private final TestHazelcastFactory hazelcastFactory = new TestHazelcastFactory();
```

Comment:
@pveentjer @mdogan These sound great but I want to this improvements if I have spare time. Lets not make these improvements a must for this pr .